lib/ansible/modules/storage/netapp/netapp_e_storagepool.py

```diff
@@ -125,7 +125,8 @@ class GroupBy(object):
     # python 2, 3 generic grouping.
     def __init__(self, iterable, key=None):
         if key is None:
-            key = lambda x: x
+            def key(x):
+                return x
```

Comment:
Here's how I'd fix this (since pylint complains about the change that autopep8 makes):

``` diff
diff --git a/lib/ansible/modules/storage/netapp/netapp_e_storagepool.py b/lib/ansible/modules/storage/netapp/netapp_e_storagepool.py
index cafcbdbbc5..c55e8132a4 100644
--- a/lib/ansible/modules/storage/netapp/netapp_e_storagepool.py
+++ b/lib/ansible/modules/storage/netapp/netapp_e_storagepool.py
@@ -120,14 +120,13 @@ def select(predicate, iterable):
         if predicate(x):
             yield x
 
+def _identity(obj):
+    return obj
 
 class GroupBy(object):
     # python 2, 3 generic grouping.
     def __init__(self, iterable, key=None):
-        if key is None:
-            def key(x):
-                return x
-        self.keyfunc = key
+        self.keyfunc = key if key else _identity
         self.it = iter(iterable)
         self.tgtkey = self.currkey = self.currvalue = object()
```